src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -27,6 +27,93 @@ const convertName = (name) => {
+    retries: framework.status.retryPolicyStatus.totalRetriedCount,
+    retryDetails: {
+      user: framework.status.retryPolicyStatus.accountableRetriedCount,
+      platform: framework.status.retryPolicyStatus.totalRetriedCount - framework.status.retryPolicyStatus.accountableRetriedCount,
+      resource: 0,
+    },
+    createdTime: new Date(framework.status.startTime).getTime(),
+    completedTime: new Date(framework.status.completionTime).getTime(),
+    appExitCode: framework.status.attemptStatus.completionStatus ? framework.status.attemptStatus.completionStatus : null,
+    virtualCluster: 'unknown',
+    totalGpuNumber: 0, // TODO
+    totalTaskNumber: framework.status.attemptStatus.taskRoleStatuses.reduce(
+      (num, statuses) => num + statuses.taskStatuses.length, 0),
+    totalTaskRoleNumber: framework.status.attemptStatus.taskRoleStatuses.length,
+  };
+};
+
+const convertTaskDetail = (taskStatus) => {
+  const completionStatus = taskStatus.attemptStatus.completionStatus;
+  return {
+    taskIndex: taskStatus.index,
+    taskState: taskStatus.state,
+    containerId: taskStatus.attemptStatus.podName,
+    containerIp: taskStatus.attemptStatus.podHostIP,
+    containerPorts: {}, // TODO
+    containerGpus: 0, // TODO
+    containerLog: '',
+    containerExitCode: completionStatus ? completionStatus.code : null,
+  };
+};
+
+const convertFrameworkDetail = (framework) => {
+  const completionStatus = framework.status.attemptStatus.completionStatus;
+  const detail = {
+    name: framework.metadata.name,
+    jobStatus: {
+      username: 'unknown',
+      state: framework.status.state,
+      subState: framework.status.state,
+      executionType: framework.spec.executionType,
+      retries: framework.status.retryPolicyStatus.totalRetriedCount,
+      retryDetails: {
+        user: framework.status.retryPolicyStatus.accountableRetriedCount,
+        platform: framework.status.retryPolicyStatus.totalRetriedCount - framework.status.retryPolicyStatus.accountableRetriedCount,
+        resource: 0,
+      },
+      createdTime: new Date(framework.status.startTime).getTime(),
+      completedTime: new Date(framework.status.completionTime).getTime(),
+      appId: framework.status.attemptStatus.instanceUID,
+      appProgress: completionStatus ? 1 : 0,
+      appTrackingUrl: '',
+      appLaunchedTime: new Date(framework.status.startTime).getTime(),
+      appCompletedTime: new Date(framework.status.completionTime).getTime(),
+      appExitCode: completionStatus ? completionStatus.code : null,
+      appExitSpec: {}, // TODO
+      appExitDiagnostics: completionStatus ? completionStatus.diagnostics : null,
+      appExitMessages: {
+        container: null,
+        runtime: null,
+        launcher: null,
+      },
+      appExitTriggerMessage: completionStatus ? completionStatus.diagnostics : null,
+      appExitTriggerTaskRoleName: null, // TODO
+      appExitTriggerTaskIndex: null, // TODO
```

Comment:
@yqwang-ms Could framework controller return some structured exit information along with exit diagnostics,
e.g. `"TaskRole master: FailedTaskCount 1 has reached MinFailedTaskCount 1: Triggered by Task: [Index: 0, Diagnostics: Pod failed with non-zero container exit code: [Container main, ExitCode: 127, Reason: ContainerCannotRun, Message: ...]]"`?

Reply:
Yep, we can do it in future. (Not first stage)